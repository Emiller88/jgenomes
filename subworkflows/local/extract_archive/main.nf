//
// extract reference genome files
//

include { GUNZIP } from '../../../modules/nf-core/gunzip'
include { UNTAR  } from '../../../modules/nf-core/untar'
include { UNZIP  } from '../../../modules/nf-core/unzip'

workflow EXTRACT_ARCHIVE {
    take:
    extract_gz
    extract_tar
    extract_zip

    main:
    versions = Channel.empty()

    // extract reference
    GUNZIP(extract_gz)
    UNTAR(extract_tar)
    UNZIP(extract_zip)

    extracted = Channel
        .empty()
        .mix(
            GUNZIP.out.gunzip,
            UNTAR.out.untar,
            UNZIP.out.unzipped_archive,
        )

    versions = versions.mix(GUNZIP.out.versions)
    versions = versions.mix(UNTAR.out.versions)
    versions = versions.mix(UNZIP.out.versions)

    emit:
    extracted // channel: [ meta, extracted_archive ]
    versions  // channel: [ versions.yml ]
}
