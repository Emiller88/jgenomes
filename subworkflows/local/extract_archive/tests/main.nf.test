nextflow_workflow {

    name "Test EXTRACT_ARCHIVE"
    script "../main.nf"
    workflow "EXTRACT_ARCHIVE"

    test(".gz && .tar.gz") {

        when {
            params {
            }
            workflow {
                """
                input[0] = Channel.of([
                        [id:'test'],
                        params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz'
                    ],
                    [
                        [id:'test'],
                        params.modules_testdata_base_path + 'generic/tar/hello.tar.gz'
                    ]
                )
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }
}
