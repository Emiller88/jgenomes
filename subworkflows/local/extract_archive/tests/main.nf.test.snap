{
    ".gz && .tar.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "hello.txt:md5,e59ff97941044f85df5297e1c302d260"
                        ]
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,078c618a24b59237a3e2257741bee7dd",
                    "versions.yml:md5,7fffc1e63ca7fcde5e4cf4e41a1c8837"
                ],
                "extracted": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "hello.txt:md5,e59ff97941044f85df5297e1c302d260"
                        ]
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "not_extracted": [
                    
                ],
                "versions": [
                    "versions.yml:md5,078c618a24b59237a3e2257741bee7dd",
                    "versions.yml:md5,7fffc1e63ca7fcde5e4cf4e41a1c8837"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.01.0"
        },
        "timestamp": "2025-03-06T13:25:51.789340756"
    }
}